packages/create-cloudflare/src/common.ts

```diff
@@ -135,32 +136,63 @@ export const setupProjectDirectory = (args: C3Args) => {
 	return { name: pathBasename, path };
 };
 
-export const offerToDeploy = async (ctx: PagesGeneratorContext) => {
+export const offerToDeploy = async (ctx: C3Context) => {
 	startSection(`Deploy with Cloudflare`, `Step 3 of 3`);
 
+	// Coerce no-deploy if it isn't possible (i.e. if its a worker with any bindings)
+	if (!(await isDeployable(ctx))) {
+		ctx.args.deploy = false;
+	}
+
 	const label = `deploy via \`${quoteShellArgs([
 		npm,
 		"run",
-		...(ctx.framework?.config.deployCommand ?? ["deploy"]),
+		...(ctx.template.deployCommand ?? ["deploy"]),
 	])}\``;
 
-	ctx.args.deploy = await processArgument(ctx.args, "deploy", {
+	const shouldDeploy = await processArgument(ctx.args, "deploy", {
 		type: "confirm",
 		question: "Do you want to deploy your application?",
 		label,
 		defaultValue: C3_DEFAULTS.deploy,
 	});
 
-	if (!ctx.args.deploy) return;
+	if (!shouldDeploy) {
+		return false;
+	}
+
+	// initialize a deployment object in context
+	ctx.deployment = {};
 
 	const loginSuccess = await wranglerLogin();
 	if (!loginSuccess) return;
```

Comment:
Should return false?

```suggestion
	if (!loginSuccess) return false;
```